homeassistant/components/traccar/device_tracker.py

```diff
@@ -318,8 +322,9 @@ async def import_events(self):
                     ),
                     None,
                 )
+                event_types = {camelcase(evt): evt for evt in EVENTS}
```

Comment:
We only want this if only `all_events` is defined as the only event.

Let's do this in the init instead:

```python
if len(event_types) == 1 and event_types[0] == EVENT_ALL_EVENTS:
    event_types == EVENTS
self._event_types = {camelcase(evt): evt for evt in event_types}
```

and then return here if the event is not in `self._event_types`

Reply:
In this case, I think we should always extend `self._event_types` when `all_events` is in the `event_types` list (not only as the only event).
The Traccar API returns all events for `["allDevices"]` and  `["ignitionOn", "allDevices"]`.

Reply:
There is no reason to do this but it's possible. If the user sets this event combination, the component will fire `traccar_None` for all not listed events. I think it's better to handle this case and be compatible with the behavior of the `API.get_events` function.